common/errors.go

```diff
@@ -1945,8 +1967,39 @@ func HasErrorCode(err error, codes ...ErrorCode) bool {
 	return false
 }
 
+func IsRetryableTowardNetwork(err error) bool {
+	// Missing data errors -> Retry
+	if HasErrorCode(err, ErrCodeEndpointMissingData) {
+		return true
+	}
+
+	// If the error says it's explicitly retryable/not retryable towards network
+	if se, ok := err.(StandardError); ok {
+		if rt, ok := se.DeepSearch("retryableTowardNetwork").(bool); ok {
+			return rt
+		}
+	}
+
+	// For exhausted upstreams, check if any of the underlying errors are retryable towards upstream
+	if HasErrorCode(err, ErrCodeUpstreamsExhausted) {
+		if exher, ok := err.(*ErrUpstreamsExhausted); ok {
+			errs := exher.Errors()
+			if len(errs) > 0 {
+				for _, e := range errs {
+					if IsRetryableTowardsUpstream(e) {
+						return true
+					}
+				}
+			}
+		}
+	}
```

Comment:
Shouldn't this be inside IsRetryableTowardsUpstream ?

Reply:
in the current logic, we are checking this against network scope

https://github.com/erpc/erpc/blob/671178233890f88c0d1918d2b209eeec2bba214b/upstream/failsafe.go#L323

Reply:
That was probably a different reason (maybe cuz we didn't expect this code for upstream errors).

Logically it does seem like a IsRetryableTowardsUpstream case, what do you think?